org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporter.java

```diff
@@ -107,13 +114,44 @@ private void importDir(Path rootFolder, IProgressMonitor monitor) {
 	}
 
 	public static GradleDistribution getGradleDistribution(Path rootFolder) {
-		if (JavaLanguageServerPlugin.getPreferencesManager() != null && JavaLanguageServerPlugin.getPreferencesManager().getPreferences().isGradleWrapperEnabled() && Files.exists(rootFolder.resolve("gradlew"))) {
-			return GradleDistribution.fromBuild();
+		PreferenceManager preferencesManager = JavaLanguageServerPlugin.getPreferencesManager();
+		if (preferencesManager != null && preferencesManager.getPreferences().isGradleWrapperEnabled() && Files.exists(rootFolder.resolve("gradlew"))) {
+			WrapperValidator validator = new WrapperValidator();
+			try {
+				ValidationResult result = validator.checkWrapper(rootFolder.toFile().getAbsolutePath());
+				if (result.isValid()) {
+					WrapperGradleDistribution gradleDistribution = GradleDistribution.fromBuild();
+					return gradleDistribution;
+				} else {
+					if (!WrapperValidator.getDisallowed().contains(result.getChecksum())) {
+						ProjectsManager pm = JavaLanguageServerPlugin.getProjectsManager();
+						if (pm != null && pm.getConnection() != null) {
+							if (preferencesManager.getClientPreferences().isActionableNotificationSupported()) {
```

Comment:
this is not sufficient. Some clients may support actionable notifications, but we don't know if they support gradle wrapper checksum management too.

I'd suggest to test for `"gradleChecksumWrapperPromptSupport":"true"` instead. If it's not supported, then do something like:

```java
//@formatter:off
String message = GRADLE_WRAPPER_CHEKSUM_WARNING_TEMPLATE.replaceAll("@wrapper@", result.getWrapperJar())
																						.replaceAll("@checksum@", result.getChecksum());
//@formatter:on
pm.getConnection().showMessage(new MessageParams(MessageType.Error, message));
```

With 
```java
	//@formatter:off
	public static final String GRADLE_WRAPPER_CHEKSUM_WARNING_TEMPLATE =
			"Security Warning! The gradle wrapper '@wrapper@' could be malicious. "
			+ "If you trust it, please add \n"
			+ "`{\"sha256\": \"@checksum@\","
			+ "\n\"allowed\": true}`"
			+ "\n to the `java.import.gradle.wrapper.checksums` preference."
			+ ""
			.replaceAll("\n", System.lineSeparator());
	//@formatter:on
```

Reply:
Fixed.